### ISSUE TITLE
Mouse Bleeds to Other Screen in Full Screen

### ISSUE THREAD

**Issue by knives-knife** [mgth/LittleBigMouse#363]
When some apps are in full-screen or borderless window mode, the mouse can still "escape" the screen/window. While the mouse is idle, the mouse is properly locked within the app. However, if the mouse is actively being pushed against the edge of the full-screen app, the mouse continues to the next monitor and then jumps back to the full-screen app. 

This most prominently occurs in games (e.g., Rainbow Six Siege as depicted in the gif attached). 
![Mouse Bleed](https://github.com/mgth/LittleBigMouse/assets/93448980/35da075b-3a7c-4115-8d26-4f6b965e4902)

**Comment by rumkugel13** [comment]
I got the same issue playing The First Descendant, thought it had something to do with the game. Anyway, when the mouse jumps to the other screen, the game loses focus momentarily and i can't shoot until i focus the game again.

**Comment by mgth** [comment]
You can add game process to excluded list in config. Does it help ?

**Comment by rumkugel13** [comment]
Yeah, seems to work. Thanks

**Comment by knives-knife** [comment]
> You can add game process to excluded list in config. Does it help ?

I went to add GTAV to the exclude list and saw that I had already added Rainbow Six Siege. So apparently I already tried that and just forgot. Also that means that the gif I posted was with Rainbow Six Siege in the exclude list. I also tried adding GTAV to the exclude list just to make sure it wasn't an issue with Rainbow Six Siege and the issue still persists. I even restarted my computer and let LittleBigMouse auto-start just in case I affected somehow while adding the process to the exclude list, but the issue still persists.

**Comment by Elduris** [comment]
I have the same issue with another game and I went as far back as version 5.1 where I remember the process exclusion worked and now it doesn't anymore. What's weird is that in the current version I can add a different program, like Chrome for example, to the excluded list and it'll work, but on 5.1 even that doesn't. Could a Windows Update have broken something?

**Comment by Soul-Vernhyde** [comment]
Yeah this makes LBM basically unusable if you play games regularly especially in shooters where it makes the camera jerk around cause it is snapping the mouse back to the center of the screen when it bleeds onto the side monitor. It really sucks at the moment.

**Comment by Emusnoc1234** [comment]
>I have the same issue with another game and I went as far back as version 5.1 where I remember the process exclusion worked and now it doesn't anymore. What's weird is that in the current version I can add a different program, like Chrome for example, to the excluded list and it'll work, but on 5.1 even that doesn't. Could a Windows Update have broken something?

Had the same problem. Switching the mouse algo to corner crossing fixed it for me.

**Comment by FatheredPuma81** [comment]
> You can add game process to excluded list in config. Does it help ?

Please add a better fix. Expecting users to add each and every single one of their games to the Exclude list is a bit much...

**Comment by BalloonSC** [comment]
> Had the same problem. Switching the mouse algo to corner crossing fixed it for me.

This works, BUT, you have to restart LBM to fix the issue for some reason. (At least for me). Even if you already have it set to corner, then quickly set it to straight, apply, then switch back and apply, it will be messed up until you restart LBM.

I just wanted to mention that because I spent so long trying to figure out what the heck was happening.

**Comment by hardenedcotton** [comment]
I have the same problem with any games that takes exclusive use of the mouse. Excluding games didnt work for me. Even after pc restart, the excluded games behaved abnormally by cursor leaving on the edges of the screen. Nothing different than what's mentioned in this thread. 

However I think I found out something different on this behavior.
I have my second display to the left and slightly above my main monitor. When in games mouse always extruded on the top left corner. afaik most games just lock the cursor to the center of the display. 

But in this case it looks like the mouse is actually closer to the left of my screen because it takes very little effort to snap out of the display. My theory is instead of games locking the cursor to the center of the display, they try lock it to the **centroid of both display's bounding box**. It's just my theory though but thought that it might be useful for devs

**Comment by FatheredPuma81** [comment]
> I have the same problem with any games that takes exclusive use of the mouse. Excluding games didnt work for me. Even after pc restart, the excluded games behaved abnormally by cursor leaving on the edges of the screen. Nothing different than what's mentioned in this thread.

Yea this has been my experience a lot recently and what you said makes a lot of sense but idk why it would only be an issue with LBM enabled. Close to just uninstalling even though I love the program for Desktop use >.>

**Comment by UnsignedV0id** [comment]
I'm having a similar issue, so I'll go a little further on my comment.

LBM conflicts with anything that locks the cursor to a monitor area; in my case, it's with Dual Monitor Tools (DMT).
I have DMT set up so my cursor is only free to move between monitors if SHIFT is held down; so, LBM will flicker my cursor between monitors when near any edge of my main screen.

I've tried to solve this with an AHK2 script that I mangled together, but with no luck; the script is not fast enough, fixes the error, but loses LBM functionality.

I'll leave the script below if anyone wants to use it, but the true ideal solution, IMO, would be a toggle key inside LBM, "only activate when 'key' is pressed down" .

[AHK Script](https://gist.github.com/UnsignedV0id/634bf918763612ea28d5d7d2803cb953)

**Comment by mgth** [comment]
👋 **[LittleBigMouse 5.3.0-beta.1](https://github.com/mgth/LittleBigMouse/releases/tag/v5.3.0-beta.1) is out and should fix this.**

LBM now detects when a game captures the mouse (hidden or confined cursor) and pauses itself automatically, resuming when you leave the game (#465 by @funnything1, optimized in #467 so high polling rate mice are not slowed down). A new **"Game detection interval"** setting is available in the layout options.

⚠️ The beta requires the [.NET 10 Desktop Runtime (x64)](https://dotnet.microsoft.com/en-us/download/dotnet/10.0) — install it first if the app doesn't start.

Could you give it a try with your games and report back? If the problem is gone, this issue will be closed with the stable release.

**Comment by mgth** [comment]
No news since the re-test request two weeks ago — closing as fixed by the 5.3.0/5.4.0 engine fixes, as announced. If you still see this on [5.4.1](https://github.com/mgth/LittleBigMouse/releases/tag/v5.4.1), just leave a comment and it reopens.
